KLSadd_insertion/src/updateChapters.py

```diff
@@ -1,110 +1,327 @@
+    """
+    hyper_headers_chap = []
+    hyper_subs_chap = []
+
+    special_input = 0
+    name_chap = word.lower()
+    print name_chap
+    if name_chap == "orthogonality":
+        special_input = 1
+    elif name_chap == "special value":
+        special_input = 2
+    elif name_chap == "symmetry":
+        special_input = 3
+    elif name_chap == "bilateral generating function":
+        special_input = 4
+
+    khyper_header_chap = []
+    k_hyper_sub_chap = []
+    index = 0
+
+    chapterstart = False
+    for item in kls:
+        index += 1
+        line = str(item)
+        if "\\subsection" in line:
+            temp = line[line.find(" ", 12)+1: line.find("}", 12)]  # get just the name (like mathpeople)
+            if temp.lower() == 'wilson':
+                chapterstart = True
+        if special_input in (0, 2, 4) and name_chap in line.lower() and chapterstart and ("\\paragraph{" in line or "\\subsubsection*{" in line) or \
+        (special_input in (1, 3) and "orthogonality relation" not in line.lower() and name_chap in line.lower() and chapterstart
+         and ("\\paragraph{" in line or "\\subsubsection*{" in line)):
+            for item in klsaddparas:
+                if index < item:
+                    klsloc = klsaddparas.index(item)
+                    break
+            t = ''.join(kls[index: klsaddparas[klsloc]])
+            k_hyper_sub_chap.append(t)  # append the whole paragraph, pray every paragraph ends with a % comment
+            khyper_header_chap.append(temp)  # append the name of subsection
+
+    for item in khyper_header_chap:
+        if item == "Pseudo Jacobi (or Routh-Romanovski)":
+            khyper_header_chap[khyper_header_chap.index(item)] = "Pseudo Jacobi"
+    k_hyp_index_iii = 0
+    offset = 0
+
+    item = 0
+    while item < len(khyper_header_chap):
+        try:
+            if khyper_header_chap[item] == khyper_header_chap[item + 1] and "generating functions" in name_chap:
+                k_hyper_sub_chap[item + 1] = k_hyper_sub_chap[item] + "\paragraph{\\bf KLS Addendum: Bilateral generating functions}" + k_hyper_sub_chap[item + 1]
+                khyper_header_chap[item] = "memes"
+                k_hyper_sub_chap[item] = "memes"
+            elif khyper_header_chap[item] == khyper_header_chap[item+1]:
+                k_hyper_sub_chap[item + 1] = k_hyper_sub_chap[item] + "\paragraph{\\bf KLS Addendum: " + \
+                    word + "}" + k_hyper_sub_chap[item + 1]
+                khyper_header_chap[item] = "memes"
+                k_hyper_sub_chap[item] = "memes"
+            else:
+                item += 1
+        except IndexError:
+            item += 1
+
+    a = 0
+    while a < len(khyper_header_chap):
+        if khyper_header_chap[a] == "memes":
+            del khyper_header_chap[a]
+            del k_hyper_sub_chap[a]
+        else:
+            a += 1
+
+    #global sorter_check
+    chap9 = 0
+
+    if sorter_check[sortloc][0] == 0:
```

Comment:
There has to be a better way to do this other than using a global sorter check - if you can't come up with it, talk to someone else more familiar with the code, but this sorter check variable is definitely not the best way to go